security/oauth1-client/src/main/java/org/glassfish/jersey/client/oauth1/RequestUtil.java

```diff
@@ -106,9 +105,9 @@ private RequestUtil() {
         String method = request.getMethod();
         MediaType mediaType = request.getMediaType();
 
-        // no entity, not a post or not x-www-form-urlencoded: return empty map
-        if (entity == null || method == null || !HttpMethod.POST.equalsIgnoreCase(method)
-                || mediaType == null || !mediaType.equals(MediaType.APPLICATION_FORM_URLENCODED_TYPE)) {
+        // no entity or not x-www-form-urlencoded: return empty map
+        if (entity == null || method == null
+                || !MediaType.APPLICATION_FORM_URLENCODED_TYPE.equals(mediaType)) {
```

Comment:
I think equals is wrong here. I would recommend using `MediaTypes.typeEqual` to ignore the possible charset encoding or other parameters at the end.